rpc/grpc/client/block_service.go

```diff
@@ -22,12 +40,32 @@ type LatestHeightResult struct {
 	Error  error
 }
 
+type getLatestHeightConfig struct {
+	chSize uint
+}
+
+type GetLatestHeightOption func(*getLatestHeightConfig)
+
+// GetLatestHeightChannelSize allows control over the channel size. If not used
+// or the channel size is set to 0, an unbuffered channel will be created.
+func GetLatestHeightChannelSize(sz uint) GetLatestHeightOption {
+	return func(opts *getLatestHeightConfig) {
+		opts.chSize = sz
+	}
+}
+
 // BlockServiceClient provides block information
 type BlockServiceClient interface {
+	// GetBlockByHeight attempts to retrieve the block associated with the
+	// given height.
 	GetBlockByHeight(ctx context.Context, height int64) (*Block, error)
-	// GetLatestHeight provides sends the latest committed block height to the given output
-	// channel as blocks are committed.
-	GetLatestHeight(ctx context.Context, resultCh chan<- LatestHeightResult)
+
+	// GetLatestBlock attempts to retrieve the latest committed block.
+	GetLatestBlock(ctx context.Context) (*Block, error)
+
+	// GetLatestHeight provides sends the latest committed block height to the
+	// resulting output channel as blocks are committed.
+	GetLatestHeight(ctx context.Context, opts ...GetLatestHeightOption) (<-chan LatestHeightResult, error)
```

Comment:
@andynog I've gone back on my prior recommendation of supplying the channel to the function here because it turns out that the goroutine this function spawns could, non-deterministically, attempt to write to the channel after it's been closed, causing the client to panic.

This new approach causes the `GetLatestHeight` method to manage the whole lifecycle of the channel, but still provides the caller control over the channel size (and whether it's buffered or unbuffered).